pkg/envoy/xdsutil.go

```diff
@@ -76,11 +77,24 @@ func GetAddress(address string, port uint32) *xds_core.Address {
 }
 
 // GetTLSParams creates Envoy TlsParameters struct.
-func GetTLSParams() *xds_auth.TlsParameters {
-	return &xds_auth.TlsParameters{
-		TlsMinimumProtocolVersion: xds_auth.TlsParameters_TLSv1_2,
-		TlsMaximumProtocolVersion: xds_auth.TlsParameters_TLSv1_3,
+func GetTLSParams(sidecarSpec v1alpha2.SidecarSpec) *xds_auth.TlsParameters {
+	minVersionInt := xds_auth.TlsParameters_TlsProtocol_value[sidecarSpec.TLSMinProtocolVersion]
```

Comment:
I know the validating webhook already checks that the strings defined for min/max TLS versions is valid, but for something like this that's relatively far removed from that I think it's worth checking these are valid again anyway and returning an error if they aren't. If something invalid does come through this function somehow this will evaluate to 0 (`TLS_AUTO`), but I'm not sure how important it is to protect against that.